images/kindnetd/cmd/kindnetd/masq.go

```diff
@@ -17,49 +17,42 @@ limitations under the License.
 package main
 
 import (
-	"bytes"
 	"fmt"
-	"strings"
 	"time"
 
-	utildbus "k8s.io/kubernetes/pkg/util/dbus"
-	utiliptables "k8s.io/kubernetes/pkg/util/iptables"
-	utilexec "k8s.io/utils/exec"
+	"github.com/coreos/go-iptables/iptables"
 )
 
 // NewIPMasqAgent returns a new IPMasqAgent
 func NewIPMasqAgent(ipv6 bool, noMasqueradeCIDRs []string) *IPMasqAgent {
-	execer := utilexec.New()
-	dbus := utildbus.New()
-	protocol := utiliptables.ProtocolIpv4
+	protocol := iptables.ProtocolIPv4
 	if ipv6 {
-		protocol = utiliptables.ProtocolIpv6
+		protocol = iptables.ProtocolIPv6
 	}
-	masqChain := utiliptables.Chain(masqChainName)
-	iptables := utiliptables.New(execer, dbus, protocol)
+	ipt, _ := iptables.NewWithProtocol(protocol)
```

Comment:
Do not ignore this error. This method actually returns errors.

Reply:
it returns errors if it doesn't find the `iptables`binary or is not able to find the version
https://github.com/coreos/go-iptables/blob/master/iptables/iptables.go#L99
this is running inside the `debian-iptables` image that should guarantee that

Reply:
Ignoring this is very poor practice because it may return errors for new reason in the future, and we don't know when the version requirements will change, etc.

Reply:
Good point